tensorflow/compiler/mlir/tosa/transforms/legalize_common.cc

```diff
@@ -4309,5 +4309,40 @@ llvm::Optional<Value> convertOneHotOp(PatternRewriter& rewriter, Operation* op,
       .getResult();
 }
 
+// Lowers 32-bit floating sign operator to a sequence of TOSA ops.
+llvm::Optional<Value> convertSignOp(PatternRewriter& rewriter, Operation* op,
+                                    Value input, RankedTensorType output_type) {
+  // Only support 32-bit floating point currently.
+  if (!output_type || !output_type.getElementType().isF32()) {
+    (void)rewriter.notifyMatchFailure(op,
+                                      "output type isn't a ranked f32 tensor");
+    return llvm::None;
+  }
+
+  // TOSA greater and select can both broadcast, so simply create a tensor with
+  // 1 element.
+  Value pos_one_const_op = getTosaConstTensorSingleF32(rewriter, op, 1.0f);
+  Value neg_one_const_op = getTosaConstTensorSingleF32(rewriter, op, -1.0f);
+  Value zero_const_op = getTosaConstTensorSingleF32(rewriter, op, 0.0f);
+
+  ShapedType const_type = output_type.clone(rewriter.getIntegerType(1));
+
+  auto gt_zero_op = CreateOpAndInfer<tosa::GreaterOp>(
+      rewriter, op->getLoc(), const_type, input, zero_const_op);
```

Comment:
You can create a ImplicitLocOpBuilder from the rewriter. This avoids having to pass op->getLoc() every time.

Reply:
CLA replyement of Contributer licence google. /Shortcuts/

Reply:
Cool, it is convenient. I add an overload of `CreateOpAndInfer` that accept `ImplicitLocOpBuilder ` as an argument.